File: erpnext/crm/report/email_summary/email_summary.js
// Copyright (c) 2021, Frappe Technologies Pvt. Ltd. and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Email Summary"] = {
	"filters": [
		{
			fieldname: "reference_document_type",
			label: __("Reference Document Type"),
			fieldtype: "Link",
			options: "DocType",
			default: "Lead"
		},
		{
			fieldname: "reference_document_name",
			label: __("Reference Document Name"),
			fieldtype: "Dynamic Link",
			options: "reference_document_type"
		},
		{
			fieldname: "from_date",
			label: __("From Date"),
			fieldtype: "Date",
			reqd: 1,
			default: frappe.datetime.add_months(frappe.datetime.get_today(), -12)
		},
		{
			fieldname: "to_date",
			label: __("To Date"),
			fieldtype: "Date",
			reqd: 1,
			default: frappe.datetime.get_today()
		},
		{
			fieldname: "company",
			label: __("Company"),
			fieldtype: "Link",
			options: "Company",
			default: frappe.defaults.get_user_default('company')
		},
		{
			fieldname: "frequency",
			label: __("Frequency"),
			fieldtype: "Select",
			options: [
				"Monthly",
				"Quarterly",
				"Half-Yearly",
				"Yearly"
			],
			default: "Monthly"
		},
		{
			fieldname: "type",
			label: __("Type"),
			fieldtype: "Select",
			options: [
				"",
				"Sent",
				"Received"
			],
			default: ""
		},
		{
			fieldname: "email_template_used",
			label: __("Email Template Used"),
			fieldtype: "Check",
			default: 0
		},
	]
};

File: erpnext/crm/report/email_summary/email_summary.py
# Copyright (c) 2021, Frappe Technologies Pvt. Ltd. and contributors
# License: MIT. See LICENSE

import frappe
from frappe import _
from erpnext.crm.report.call_log_summary.call_log_summary import get_period_list, validate_filters, prepare_data, get_chart_data

def execute(filters=None):
	return EmailSummary(filters).run()
	columns, data = [], []
	return columns, data

class EmailSummary():
	def __init__(self, filters):
		self.filters = frappe._dict(filters or {})

	def run(self):
		validate_filters(self.filters.from_date, self.filters.to_date)
		columns = self.get_columns()
		data = prepare_data(columns, self.get_data())
		chart = get_chart_data(columns, data)
		columns.insert(0, {
            'fieldname': 'user',
            'label': _('{0} Owner').format(self.filters.reference_document_type),
            'fieldtype': 'Link',
            'options': 'User',
			'width': 200
        })
		return columns, data, None, chart

	def get_columns(self):
		return get_period_list(self.filters.from_date, self.filters.to_date, self.filters.frequency)

	def get_data(self):
		return frappe.db.sql(self.get_query(), as_dict=True, debug=1)

	def get_query(self):
		filters = self.filters

		join_field = """c.reference_doctype = '{0}' 
			AND c.reference_name = dt.name
			AND (c.creation BETWEEN '{1}' AND '{2}')
			""".format(filters.reference_document_type, filters.from_date, filters.to_date)

		if filters.reference_document_name:
			join_field += " AND c.reference_name = '{0}'".format(filters.reference_document_name)

		if filters.email_template_used:
			join_field += " AND c.email_template IS NOT NULL"

		if filters.company:
			join_field += " AND dt.company = '{0}'".format(filters.company)

		query = """SELECT c.name as name,
			DATE(c.creation) as creation, 
			dt._assign as assign,
			dt.name as reference_name
    		FROM (`tabCommunication` c 
			INNER JOIN `tab{0}` dt ON {1})
			""".format(filters.reference_document_type, join_field)

		return query